integration-tests/spring-web/src/test/java/io/quarkus/it/spring/web/SpringControllerTest.java

```diff
@@ -98,4 +98,12 @@ public void testRestControllerWithoutRequestMapping() {
         RestAssured.when().get("/hello").then()
                 .body(containsString("hello"));
     }
+
+    @Test
+    public void testResponseEntityWithIllegalArgumentException() {
+        RestAssured.when().get("/exception/re").then()
+                .contentType("application/json")
+                .body(containsString("hello from error"))
+                .statusCode(402);
```

Comment:
Out of curiosity, where does this status code come from? `402` seems pretty unusual, but maybe it's completely normal in that case.

Reply:
I just added a random one to make ensure the response was different

Reply:
It could have been anything, I just picked one in random 🤪

Reply:
Thanks for the explanation! :)

Reply:
NP!